src/battle_ai_switch_items.c

```diff
@@ -28,7 +28,45 @@ static bool8 ShouldUseItem(u32 battler);
 static bool32 AiExpectsToFaintPlayer(u32 battler);
 static bool32 AI_ShouldHeal(u32 battler, u32 healAmount);
 static bool32 AI_OpponentCanFaintAiWithMod(u32 battler, u32 healAmount);
-static bool32 IsAiPartyMonOHKOBy(u32 battlerAi, u32 battlerAtk, struct Pokemon *aiMon);
+
+static void InitializeSwitchinCandidate(struct Pokemon *mon)
```

Comment:
I still think it would be better to use PokemonToBattleMon instead of duplicating code. Could you test if the difference in the computation time is this big? I think this function only leaves out nickname and status2.

Reply:
Yep you're right! The difference used to be bigger (swing of about 0.06s), but the initial version of this PR had a very basic implementation of calculating hits to KO (just dividing health by damage of strongest move) until PCG asked if I could also consider status and entry hazards, which kinda ballooned the scope and also required me to assign more things.

The hitsToKO consideration is obviously much smarter now but we're at about 0.47s delay rather than what used to be low 0.4 - 0.43s, and some of that is eating new assignments that didn't used to be necessary.

Will switch over to PokemonToBattleMon in the next commits, I don't see a perceptible difference anymore :)